blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -473,7 +485,15 @@ private MetricData transformColumnsToMetricData(Locator locator, ColumnList<Long
                                                                        Granularity gran) {
         try {
             RollupType rollupType = RollupType.fromString(metaCache.get(locator, rollupTypeCacheKey));
-            DataType dataType = new DataType(metaCache.get(locator, dataTypeCacheKey));
+            DataType dataType = getDataType(locator, dataTypeCacheKey);
+            if (rollupType == null) {
+                rollupType = RollupType.BF_BASIC;
+            }
+
+            if (dataType == null) {
+                dataType = DataType.NUMERIC;
+            }
+
```

Comment:
I've seen this code several times already. Perhaps bake this logic into the `getDataType` and `RollupType.fromString` methods?